lib/grape/validations/validators/values.rb

```diff
@@ -7,6 +7,7 @@ def initialize(attrs, options, required, scope)
       end
 
       def validate_param!(attr_name, params)
+        return unless params.is_a?(Hash)
```

Comment:
This just skips the validation of the inner param if the outer param does not have a valid value.

The "real" error happens in the `coerce` validator that correctly invalidates a `Hash` param with string value.

This is the same thing happening in the `allow_blank` validator here: https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/validators/allow_blank.rb#L5